src/stream.rs

```diff
@@ -196,8 +204,22 @@ impl Compress {
     /// Creates a new compressor ready to encode data into brotli
     pub fn new(params: &CompressParams) -> Compress {
         unsafe {
-            let state = brotli_sys::RustBrotliCompressorCreate(params.params);
+            let state = brotli_sys::BrotliEncoderCreateInstance(None, None, 0 as *mut _);
             assert!(!state.is_null());
+
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::RUST_PARAM_MODE,
+                                                  params.mode_as_native());
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::RUST_PARAM_QUALITY,
+                                                  params.quality);
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::RUST_PARAM_LGWIN,
+                                                  params.lgwin);
+            brotli_sys::BrotliEncoderSetParameter(state,
+                                                  brotli_sys::RUST_PARAM_LGBLOCK,
+                                                  params.lgblock);
```

Comment:
It looks like the encoder can be created before setting the parameters, so the `CompressParams` struct may no longer make sense. Perhaps this could remove `CompressParams` altogether and move the various configuration methods to configuration on `Compress` directly?